code/modules/reagents/reagent_containers/food/snacks.dm

```diff
@@ -1502,6 +1503,54 @@
 	reagents.add_reagent(NUTRIMENT, 8)
 	bitesize = 2
 
+/obj/item/weapon/reagent_containers/food/snacks/gunkburger
+	name = "gunk burger"
+	desc = "A GunkCo classic! You will eat the bugs and you will enjoy them."
+	icon_state = "bugburger"
+	food_flags = FOOD_MEAT
+	base_crumb_chance = 20
+	
+/obj/item/weapon/reagent_containers/food/snacks/gunkburger/New()
+	..()
+	processing_objects += src
+	reagents.add_reagent(NUTRIMENT, 6)
+	if(prob(30))
+		reagents.add_reagent(SALTWATER, 3) //the best non-karm emetic we have
+		desc = "You can feel the legs wriggling and bug juices with each bite. Oh god, you're gonna THR-"
+	bitesize = 2
+
+/obj/item/weapon/reagent_containers/food/snacks/gunkburger/pickup(mob/user)
+	timer = 0
+
+/obj/item/weapon/reagent_containers/food/snacks/gunkburger/process()
+	timer += 1
+	if(timer > 30 && istype(loc, /turf) && prob(60))
+		Move(get_step(loc, pick(cardinal)))
+
+/obj/item/weapon/reagent_containers/food/snacks/deluxegunkburger
+	name = "deluxe gunk burger"
+	desc = "GunkCo's latest innovation! You won't guess the special ingredient!"
+	icon_state = "deluxebugburger"
+	food_flags = FOOD_MEAT
+	base_crumb_chance = 20
+
+/obj/item/weapon/reagent_containers/food/snacks/deluxegunkburger/New()
+	..()
+	processing_objects += src
+	reagents.add_reagent(NUTRIMENT, 12)
+	if(prob(30))
+		reagents.add_reagent(SALTWATER, 3)
+		desc = "You can't comprehend how much I regret biting into this thing. The disgusting texture, burning juices and terrible taste will never leave my mind."
+	bitesize = 2
+
+/obj/item/weapon/reagent_containers/food/snacks/deluxegunkburger/pickup(mob/user)
+	timer = 0
+
+/obj/item/weapon/reagent_containers/food/snacks/deluxegunkburger/process()
+	timer += 1
+	if(timer > 30 && istype(loc, /turf) && prob(60))
```

Comment:
instead of having this be separate, you could make an abstract parent that has the behavior and make these foods children of it, like how pies work

Reply:
I tried. Please check @Kurfursten.